python/tank/authentication/session_cache.py

```diff
@@ -462,6 +472,31 @@ def get_current_host():
     return host
 
 
+def _get_recent_items(document, recent_field, current_field, type_name):
+    """
+    Extract the list of recent items from the document.
+
+    If the recent_field is not set, then we'll simply return the current_field's
+    value. The recent_field will be empty when upgrading from an older core
+    that didn't support the recent users/hosts list.
+
+    :param object document: Document to extract information from
+    :param recent_field: Field from which we need to retrieve
+    """
+    # Extract the list of recent items.
+    items = document[recent_field]
```

Comment:
for extra safety, maybe `items = document.get(recent_field)`? Is this a dictionary?

Reply:
Not necessary. The code that reads the file actually fills in missing fields, if any, so the code afterwards can just assuming is properly formatted.